sdk/core/azure-core/src/test/java/com/azure/core/util/polling/OperationResourcePollingStrategyTest.java

```diff
@@ -406,8 +410,10 @@ public void operationLocationPollingStrategyServiceVersionTest(String requestPol
 
     static Stream<int[]> statusCodeProvider() {
         return Stream.of(
+            // poll 500, poll 200 Succeeded, final 200
             new int[]{500, 200, 200, 3},
-            new int[]{200, 500, 200, 2});
+            // poll 200 Succeeded, final 500, final 200
+            new int[]{200, 500, 200, 3});
```

Comment:
The test case was wrong.

In the buggy version, 1st 200 with `status=Retry` is treated as complete, hence the 2nd 500 not happen at all, then it gets to the finalUrl. Hence count=2

Updated the test to
- 1st, poll 200 with `status=Succeeded`
- poll done, 2nd, final 500, pipeline do retry
- 3nd, final 200